Salwador/Modal/ImageGenerationResponse.swift

```diff
@@ -0,0 +1,12 @@
+// Created by Florian Schweizer on 09.11.22
+
+import Foundation
+
+struct ImageGenerationResponse: Codable {
```

Comment:
I guess I need to understand the difference between the two.

Reply:
Here are two articles I found on the web that might make it clearer for you.

https://swiftbyrahul.com/2020/10/how-to-use-encodable-and-decodable-in-swift/

https://stackoverflow.com/questions/60652158/swift-encodable-decodable-vs-codable

Reply:
Good article, but app can't be built if I change it as **Encodable**.

Reply:
Oh, sorry, it's my fault. I mixed up encoding and decoding in this case. 

It needs to be `Decodable` only and not `Encodable`. Because it is the response body that you want to decode into an `ImageGenerationResponse`. Sorry for the confusion.

But we also have the case for creating a model that could conform to `Encoding` in our code. 

Currently, you are using a [dictionary in combination with the older `JSONSerialization` API](https://github.com/memeteli/Salvador/pull/1/files#diff-6475931b747b2dccead0c696cd3c7fa5139199a85339258cab7ff9a5b185711eR21) for JSON encoding of the request body. 

Instead, you could create another model called `ImageGenerationRequest` that conforms to `Encoding` which would represent the data structure of your currently used dictionary. Create an instance of it and use `JSONEncoder` instead of `JSONSerialization`  to serialize it.